web/src/features/AppConfig/components/AppConfig.tsx

```diff
@@ -606,7 +606,7 @@ class AppConfig extends Component<Props, State> {
         const changed = this.isConfigChanged(newGroups);
         this.setState({ configGroups: newGroups, changed });
         if (this.props.isEmbeddedCluster) {
-          this.props.setNavbarConfigGroups({ newGroups, changed });
```

Comment:
This was the actual problem, we were passing an object instead of an array of config group.